md/enhanced.py

```diff
@@ -445,23 +442,29 @@ def generate_solvent_phase_samples(
     u_interval = steps_per_batch
     x_interval = steps_per_batch
 
-    all_xs = []
-
-    for iteration in range(num_batches):
-        _, _, _ = ctxt.multiple_steps_U(lamb, num_steps, lambda_windows, u_interval, x_interval)
-        x_t = ctxt.get_x_t()
-        observable_fn(iteration, x_t)
-        v_t = ctxt.get_v_t()
-        box = ctxt.get_box()
-        all_xs.append(x_t)
-        if mc_mover is not None:
-            xvb = mc_mover.move(CoordsVelBox(x_t, v_t, box))
-            ctxt.set_x_t(xvb.coords)
-            # don't have setters for x_t and box right now
-            np.testing.assert_array_equal(xvb.velocities, v_t)
-            np.testing.assert_array_equal(xvb.box, box)
-
-    return np.ndarray(all_xs)
+
+    class Hackery:
```

Comment:
This is a horrible hack, but it demonstrates what @proteneer was trying to avoid.

We have to keep all of the impls referenced somewhere, else GC cleans them up and the context object seg faults.